src/coreclr/src/vm/olevariant.cpp

```diff
@@ -300,7 +300,7 @@ VARTYPE OleVariant::GetVarTypeForTypeHandle(TypeHandle type)
         return GetVarTypeForCVType(CorElementTypeToCVTypes(elemType));
 
     // Handle objects.
-    if (!type.IsTypeDesc())
+    if (!type.IsArray())
```

Comment:
This should also exclude TypeDescs as before.

Reply:
Right. Good catch!

---
In reply to: [362655649](https://github.com/dotnet/runtime/pull/1201#discussion_r362655649) [](ancestors = 362655649)